tests/components/sensor/test_pilight.py

```diff
@@ -0,0 +1,117 @@
+        self.hass.config.components = ['pilight']
+
+    def tearDown(self):  # pylint: disable=invalid-name
+        """Stop down everything that was started."""
+        self.hass.stop()
+
+    def test_sensor_value_from_code(self):
+        """Test the setting of value via pilight."""
+        assert _setup_component(self.hass, sensor.DOMAIN, {
+            sensor.DOMAIN: {
+                'platform': 'pilight',
+                'name': 'test',
+                'variable': 'test',
+                'payload': {'protocol': 'test-protocol'},
+                'unit_of_measurement': 'fav unit'
+            }
+        })
+
+        # Set value from data with correct payload
+        fire_pilight_message(hass=self.hass,
+                             protocol='test-protocol',
+                             data={'test': 42})
+        self.hass.block_till_done()
+        state = self.hass.states.get('sensor.test')
+        self.assertEqual('42', state.state)
+
+        # Check if unit is set properly
+        self.assertEqual('fav unit',
+                         state.attributes.get('unit_of_measurement'))
+
+    def test_disregard_wrong_payload(self):
+        """Test omitting setting of value with wrong payload."""
+        assert _setup_component(self.hass, sensor.DOMAIN, {
+            sensor.DOMAIN: {
+                'platform': 'pilight',
+                'name': 'test',
+                'variable': 'test',
+                'payload': {'uuid': '1-2-3-4',
+                            'protocol': 'test-protocol'}
+            }
+        })
+
+        # Try set value from data with incorrect payload
+        fire_pilight_message(hass=self.hass,
+                             protocol='test-protocol',
+                             data={'test': 'data',
+                                   'uuid': '0-0-0-0'})
+        self.hass.block_till_done()
+        state = self.hass.states.get('sensor.test')
+        self.assertEqual('unknown', state.state)
+
+        # Try set value from data with partially matched payload
+        fire_pilight_message(hass=self.hass,
+                             protocol='wrong-protocol',
+                             data={'test': 'data',
+                                   'uuid': '1-2-3-4'})
+        self.hass.block_till_done()
+        state = self.hass.states.get('sensor.test')
+        self.assertEqual('unknown', state.state)
+
+        # Try set value from data with fully matched payload
+        fire_pilight_message(hass=self.hass,
+                             protocol='test-protocol',
+                             data={'test': 'data',
+                                   'uuid': '1-2-3-4',
+                                   'other_payload': 3.141})
+        self.hass.block_till_done()
+        state = self.hass.states.get('sensor.test')
+        self.assertEqual('data', state.state)
+
+    @unittest.SkipTest
+    @patch('homeassistant.components.sensor.pilight._LOGGER')
```

Comment:
I thought so. It is not clear to me why the logging mock does not work here (error mock does not work too), but in other unit tests it does work. Without TestCase setup and teardown methods are more ugly to my taste, but I can do it that way.